### DISCUSSION TITLE
Can spglib detect not-necessary super-cell ?

### DISCUSSION THREAD

**Discussion by husakm** [spglib/spglib#455] [Q&A]
Just a question:
Can spglib detect not necessary super-cell ?
I mean to detect situation: I had chosen some lattice parameter e.g. 2x bigger than necessary ...
I was trying to do this on test data by the spg_standardize_cell  + large symprec call. It looks like spglib does not have such functionality ...
It was returning the original cell as standard ...

**Comment by atztogo** [comment]
I can not see what you mean. To clarify, providing an example is helpful.

**Reply by LecrisUT** [reply]
I think that is either `primitive_lattice` or `std_lattice`. Can you clarify the difference between these 2 here? It would help with the C++ wrapper as well.

**Reply by atztogo** [reply]
Not only those. When lattice point group type of supercell lattice is different from that of the primitive lattice, a special treatment is applied to the matrix representations of the space group operations of the supercell. However, I am not very sure if @husakm is mentioning this or not. A python script as an example is the best way to see it for me.

**Reply by LecrisUT** [reply]
The point about the point group would be good to document (most likely already is, but it should be more visible). Naively I am only considering the case that @husakm is mentioning that you simply shrink the unit cell and I would not expect that there is some nuance beyond that.

Probably this is covered on [this page](https://spglib.readthedocs.io/en/latest/definition.html#spglib-conventions-of-standardized-unit-cell)?

**Reply by atztogo** [reply]
Often questions are not well defined or sentences without mathematical formulae are not suitable for describing the questions. But fortunately, we can communicate via a python script or C code, which is more concrete and can be a shortcut. Why not using it?

In addition, we wrote a manuscript https://arxiv.org/abs/1808.01590, where you can find the details. Users can read it in much shorter time than the time I spent for writing it.

**Reply by LecrisUT** [reply]
Sure, but the user might not have the background or vocabularly to express what they don't know that are nuances. Fleshing these out with non-experts + examples like [this ones](https://spglib.readthedocs.io/en/latest/definition.html#transformation-to-a-primitive-cell) will help make the documentation more navigable and intuitive. For example, we could improve them, using a more intuitive and simple example crystal structure.

**Comment by husakm** [comment]
OK. I will be more illustrative:
Let expect I had made an incorrect choice of an orthorhombic unit cell.
I had chose a cell a=30,b=5,c=3.
It is a supercell, de facto a combination of 2 identical cells (A,B) in a direction.
The correct cell should be:
a=15,b=5,c=3.
To describe only the A part will be fully sufficent. I ask whatewer spglib can detect such situation and offer a standard cell
as A with the correct lattice parameters and 0.5 cell volume.
Such incorrect cell choice happen in reality for triclinic system. The simple orthorhombic is only an ilustration.
i can give sample data as C source code for such structure if it give sense.

**Reply by atztogo** [reply]
You illustration is not precise enough. Your source code example is necessary.

**Reply by husakm** [reply]
I can generated data from the real live CIF files with such issues. But the structures are big.
In other words:
I chose incorrectly a cell 2x or  4 x  bigger than necessary.
It means that I had chosen a super-cell instead of the standard cell.
The correct cell have e.g. 1/2 parameter a in comparison tot he supercell.
I want to know whatever spglib can detect such situation and find the smaller cell.
Do you understand finally ?

**Reply by atztogo** [reply]
spglib can find the smaller cell.

**Reply by LecrisUT** [reply]
[this test](https://github.com/spglib/spglib/blob/a66056afbe8434cecd7a127aa3ee7290d4a42b0b/test/functional/c/test_find_primitive_cell.cpp#L23) is probably showing that.

Although I think that interface should be deprecated since the same information seems to be available on the `SpglibDataset`. I couldn't mark it as deprecated because it is doing further operations after getting the dataset, I didn't look to carefully on de-tangling it.

**Comment by husakm** [comment]
Hi

I can confirm SPGLIB works OK as expected. The sample bellow shows an artificial example of super-cell (created by me from a simple structure in P-1, CSD code ABAHIW). The lattice parameters `a` is 2x bigger than necessary. Call to `spg_standardize_cell` gives correctly a cell with 52 atoms (the original one is 104).  I must check now why it does not work with real live non-simulated experimental data  incorrectly described by authors in super-cell ... Maybe it will require some extreme tolerance to atom miss-placement.

<details>

```c
//section  generated by checkCIF-DFT
int num_atom =104;
double lattice[3][3] = {
{6.72900009155273,-0.739408671855927,-0.90732330083847},
{0,7.15388965606689,-1.82373034954071},
{0,0,20.6759014129639}
};

double position[104][3] = {
{0.245039999485016,0.352299988269806,0.0644000023603439},
{0.301600009202957,0.129960000514984,0.131060004234314},
{0.202299997210503,0.790139973163605,0.385870009660721},
{0.23819999396801,0.00938999652862549,0.321410000324249},
{0.227799996733665,0.844399988651276,0.332170009613037},     // Nitrogen
{0.253500014543533,0.384499996900558,0.238749995827675},
{0.254000008106232,0.255899995565414,0.24609999358654},
{0.24210000038147,0.517599999904633,0.290369987487793},
{0.232500001788139,0.481799989938736,0.333249986171722},
{0.245000004768372,0.703199982643127,0.2778500020504},
{0.258599996566772,0.762700021266937,0.21620999276638},
{0.261799991130829,0.89190000295639,0.209299996495247},
{0.267199993133545,0.627499997615814,0.164879992604256},
{0.275200009346008,0.663800001144409,0.122050002217293},
{0.264200001955032,0.439300000667572,0.175990000367165},
{0.273200005292892,0.289000004529953,0.122160002589226},
{0.239099994301796,0.216700002551079,0.00693000014871359},
{0.20270000398159,0.0869999974966049,0.0196000002324581},
{0.443199992179871,0.229300007224083,0.979860007762909},
{0.476599991321564,0.355100005865097,0.965900003910065},
{0.443500012159348,0.135800004005432,0.942749977111816},
{0.542400002479553,0.205200001597404,0.0132499998435378},
{0.0768999978899956,0.268900007009506,0.961740016937256},
{0.950299978256226,0.263799995183945,0.983449995517731},
{0.0623999983072281,0.180600002408028,0.922850012779236},
{0.112499997019768,0.396299988031387,0.949500024318695},
{0.754960000514984,0.647700011730194,0.435600012540817},
{0.698400020599365,0.870039999485016,0.368939995765686},
{0.797699987888336,0.209859997034073,0.11412999778986},
{0.761799991130829,0.990610003471375,0.178589999675751},
{0.772199988365173,0.155599996447563,0.167830005288124},
{0.746500015258789,0.615499973297119,0.261249989271164},
{0.745999991893768,0.744099974632263,0.253899991512299},
{0.75789999961853,0.482400000095367,0.209629997611046},
{0.767499983310699,0.518199980258942,0.166749998927116},
{0.754999995231628,0.29679998755455,0.2221499979496},
{0.741400003433228,0.237299993634224,0.283789992332458},
{0.738200008869171,0.10809999704361,0.290699988603592},
{0.732800006866455,0.372500002384186,0.335119992494583},
{0.724799990653992,0.336199998855591,0.377950012683868},
{0.73580002784729,0.560699999332428,0.324010014533997},
{0.72680002450943,0.711000025272369,0.377840012311935},
{0.760900020599365,0.78329998254776,0.493070006370544},
{0.797299981117249,0.912999987602234,0.480399996042252},
{0.556800007820129,0.770699977874756,0.520139992237091},
{0.523400008678436,0.644900023937225,0.534099996089935},
{0.556500017642975,0.864199995994568,0.557250022888184},
{0.457599997520447,0.794799983501434,0.48675000667572},
{0.923099994659424,0.731100022792816,0.538259983062744},
{0.0497000217437744,0.736199975013733,0.516550004482269},
{0.937600016593933,0.819400012493134,0.577149987220764},
{0.887499988079071,0.60369998216629,0.550499975681305},
{0.245039999485016,0.352299988269806,0.564400017261505},
{0.301600009202957,0.129960000514984,0.631060004234314},
{0.202299997210503,0.790139973163605,0.885869979858398},
{0.23819999396801,0.00938999652862549,0.821410000324249},
{0.227799996733665,0.844399988651276,0.832170009613037},
{0.253500014543533,0.384499996900558,0.738749980926514},
{0.254000008106232,0.255899995565414,0.746100008487701},
{0.24210000038147,0.517599999904633,0.790380001068115},
{0.232500001788139,0.481799989938736,0.833249986171722},
{0.245000004768372,0.703199982643127,0.777849972248077},
{0.258599996566772,0.762700021266937,0.716210007667542},
{0.261799991130829,0.89190000295639,0.709299981594086},
{0.267199993133545,0.627499997615814,0.664879977703094},
{0.275200009346008,0.663800001144409,0.622049987316132},
{0.264200001955032,0.439300000667572,0.675989985466003},
{0.273200005292892,0.289000004529953,0.622160017490387},
{0.239099994301796,0.216700002551079,0.506919980049133},
{0.20270000398159,0.0869999974966049,0.519599974155426},
{0.443199992179871,0.229300007224083,0.479860007762909},
{0.476599991321564,0.355100005865097,0.465900003910065},
{0.443500012159348,0.135800004005432,0.442750006914139},
{0.542400002479553,0.205200001597404,0.51324999332428},
{0.0768999978899956,0.268900007009506,0.461739987134933},
{0.950299978256226,0.263799995183945,0.483449995517731},
{0.0623999983072281,0.180600002408028,0.422850012779236},
{0.112499997019768,0.396299988031387,0.449499994516373},
{0.754960000514984,0.647700011730194,0.935599982738495},
{0.698400020599365,0.870039999485016,0.868939995765686},
{0.797699987888336,0.209859997034073,0.614130020141602},
{0.761799991130829,0.990610003471375,0.678579986095428},
{0.772199988365173,0.155599996447563,0.667829990386963},
{0.746500015258789,0.615499973297119,0.761250019073486},
{0.745999991893768,0.744099974632263,0.753899991512299},
{0.75789999961853,0.482400000095367,0.709619998931885},
{0.767499983310699,0.518199980258942,0.666750013828278},
{0.754999995231628,0.29679998755455,0.722150027751923},
{0.741400003433228,0.237299993634224,0.783789992332458},
{0.738200008869171,0.10809999704361,0.790700018405914},
{0.732800006866455,0.372500002384186,0.835120022296906},
{0.724799990653992,0.336199998855591,0.877950012683868},
{0.73580002784729,0.560699999332428,0.824010014533997},
{0.72680002450943,0.711000025272369,0.877839982509613},
{0.760900020599365,0.78329998254776,0.993070006370544},
{0.797299981117249,0.912999987602234,0.980400025844574},
{0.556800007820129,0.770699977874756,0.0201400518417358},
{0.523400008678436,0.644900023937225,0.0341000556945801},
{0.556500017642975,0.864199995994568,0.0572500228881836},
{0.457599997520447,0.794799983501434,0.98675000667572},
{0.923099994659424,0.731100022792816,0.0382599830627441},
{0.0497000217437744,0.736199975013733,0.0165499448776245},
{0.937600016593933,0.819400012493134,0.0771499872207642},
{0.887499988079071,0.60369998216629,0.0505000352859497},
};

int types[104]= {
8,8,8,8,7,6,1,6,1,6,6,1,6,1,6,6,6,1,6,1,1,1,6,1,1,1,8,8,8,8,7,6,1,6,1,6,6,1,6,1,6,6,6,1,6,1,1,1,6,1,1,1,8,8,8,8,7,6,1,6,1,6,6,1,6,1,6,6,6,1,6,1,1,1,6,1,1,1,8,8,8,8,7,6,1,6,1,6,6,1,6,1,6,6,6,1,6,1,1,1,6,1,1,1,
};
//end of section  generated by checkCIF-DFT
```

</details>

**Reply by LecrisUT** [reply]
I've added `<details>` around your code-snippet to make it more readable. It would be helpful if you could annotate 2 atoms that should be marked as equivalent in the data you provided.

**Reply by husakm** [reply]
I had corrected the example and unfortunately deleted the `<details>`. I am not able to restore it ...
According your request: The structure contains only one Nitrogen atom (number 7). So it is in the structure 4 times (2x as P-1 and 2 more in the additional supercell). It is clear from the `types` what atom it is ...

**Reply by LecrisUT** [reply]
> I had corrected the example and unfortunately deleted the `<details>`. I am not able to restore it ...

syntax is like:
````markdown
<details>

```c
...
```

</details>
````

Note the extra lines.

I have edited again to comment on the line with Nitrogen, can you put a comment on the other ones as well?
